src/modules/wallet/__tests__/wallet.service.spec.ts

```diff
@@ -687,4 +691,386 @@ describe('WalletService', () => {
+        id: uuid.v4(),
+        name: 'Mock Created Wallet Name',
+        logo_url: 'http://test.com/mock-logo.png',
+        cover_url: 'http://test.com/mock-cover.png',
+        password: 'mockPassword',
+        salt: 'mockSalt',
+        created_at: new Date(),
+      };
+
+      jest
+        .spyOn(walletService, 'getByName')
+        .mockResolvedValueOnce(mockSenderWallet);
+      jest
+        .spyOn(walletService, 'createWallet')
+        .mockResolvedValue(mockCreatedWallet);
+      jest
+        .spyOn(tokenService, 'countTokenByWallet')
+        .mockResolvedValueOnce(1000);
+      jest
+        .spyOn(transferService, 'transferBundle')
+        .mockResolvedValue(undefined);
+      jest
+        .spyOn(walletService, 'addWalletToMapConfig')
+        .mockResolvedValue(undefined);
+
+      const senderWallet = 'sender_wallet';
+      const tokenTransferAmountDefault = 100;
+      const walletId = 'parent_wallet_id';
+      const csvJson = [
+        { wallet_name: 'wallet1', token_transfer_amount_overwrite: 50 },
+        { wallet_name: 'wallet2' },
+      ];
+      const filePath = 'file.csv';
+
+      const result = await walletService.batchCreateWallet(
+        senderWallet,
+        tokenTransferAmountDefault,
+        walletId,
+        csvJson,
+        filePath,
+      );
+
+      expect(result).toEqual({
+        message: 'Batch wallet creation successful',
+      });
+      expect(walletService.getByName).toHaveBeenCalledWith(senderWallet);
+      expect(walletService.createWallet).toHaveBeenCalledTimes(2);
+      expect(transferService.transferBundle).toHaveBeenCalledTimes(2);
+      expect(fs.promises.unlink).toHaveBeenCalledWith(filePath);
+    });
+
+    it('should fail when sender wallet does not exist', async () => {
+      jest.spyOn(walletService, 'getByName').mockResolvedValueOnce(null);
+      jest.spyOn(walletService, 'createWallet').mockImplementation(() => {
+        throw new Error('This should not be called');
+      });
+      jest.spyOn(transferService, 'transferBundle').mockImplementation(() => {
+        throw new Error('This should not be called');
+      });
+
+      const senderWallet = 'nonexistent_wallet';
+      const tokenTransferAmountDefault = 100;
+      const walletId = 'parent_wallet_id';
+      const csvJson = [
+        { wallet_name: 'wallet1', token_transfer_amount_overwrite: 50 },
+      ];
+      const filePath = 'file.csv';
+
+      await expect(
+        walletService.batchCreateWallet(
+          senderWallet,
+          tokenTransferAmountDefault,
+          walletId,
+          csvJson,
+          filePath,
+        ),
+      ).rejects.toThrowError('Sender wallet does not exist');
+
+      expect(walletService.getByName).toHaveBeenCalledWith(senderWallet);
+      expect(walletService.createWallet).not.toHaveBeenCalled();
+      expect(transferService.transferBundle).not.toHaveBeenCalled();
+      expect(fs.promises.unlink).toHaveBeenCalledWith(filePath);
```

Comment:
@Kpoke i've removed the check for jest ID from service logic, and added a test for deleting file here.